src/pydap/net.py

```diff
@@ -23,7 +24,8 @@ def GET(url, application=None, session=None):
 
 
 def raise_for_status(response):
-    if response.status_code >= 400:
+    # Raise error if status is above 300:
+    if response.status_code >= 300:
```

Comment:
The client already attempts (through the ``session.head`` call) to follow redirects. It is fails then an error should be rightly raised. At least that's the thinking...